zrml/market-commons/src/migrations.rs

```diff
@@ -14,3 +14,288 @@
+                creator: legacy_market.creator,
+                creation: legacy_market.creation,
+                creator_fee: legacy_market.creator_fee,
+                oracle: legacy_market.oracle,
+                metadata: legacy_market.metadata,
+                market_type: legacy_market.market_type,
+                period: legacy_market.period,
+                scoring_rule: legacy_market.scoring_rule,
+                status: legacy_market.status,
+                report: legacy_market.report,
+                resolved_outcome: legacy_market.resolved_outcome,
+                dispute_mechanism,
+                deadlines: legacy_market.deadlines,
+            };
+            new_markets_data.push((key, new_market));
+        }
+        for (key, new_market) in new_markets_data {
+            put_storage_value::<MarketOf<T>>(MARKET_COMMONS, MARKETS, &key, new_market);
+            total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        }
+        log::info!("Completed updates of markets");
+        StorageVersion::new(MARKET_COMMONS_NEXT_STORAGE_VERSION).put::<Pallet<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<(), &'static str> {
+        use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+        for (key, legacy_market) in storage_iter::<LegacyMarketOf<T>>(MARKET_COMMONS, MARKETS) {
+            Self::set_temp_storage(legacy_market, &format!("{:?}", key.as_slice()));
+        }
+        Ok(())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade() -> Result<(), &'static str> {
+        use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+        for (key, updated_market) in storage_iter::<MarketOf<T>>(MARKET_COMMONS, MARKETS) {
+            let legacy_market: LegacyMarketOf<T> =
+                Self::get_temp_storage(&format!("{:?}", key.as_slice()))
+                    .expect("legacy market not found");
+            assert_eq!(updated_market.creator, legacy_market.creator);
+            assert_eq!(updated_market.creation, legacy_market.creation);
+            assert_eq!(updated_market.creator_fee, legacy_market.creator_fee);
+            assert_eq!(updated_market.oracle, legacy_market.oracle);
+            assert_eq!(updated_market.metadata, legacy_market.metadata);
+            assert_eq!(updated_market.market_type, legacy_market.market_type);
+            assert_eq!(updated_market.period, legacy_market.period);
+            assert_eq!(updated_market.deadlines, legacy_market.deadlines);
+            assert_eq!(updated_market.scoring_rule, legacy_market.scoring_rule);
+            assert_eq!(updated_market.status, legacy_market.status);
+            assert_eq!(updated_market.report, legacy_market.report);
+            assert_eq!(updated_market.resolved_outcome, legacy_market.resolved_outcome);
+        }
+        Ok(())
```

Comment:
Could you add a test that verifies that the correct pallet version has been set?